addons/test_mail/models/test_mail_models.py

```diff
@@ -165,3 +165,36 @@ class MailTestComposerMixin(models.Model):
 
     def _compute_render_model(self):
         self.render_model = self._name
+
+
+class MailTestDocumentTags(models.Model):
+    """ Model required for mail.test.document to define a many2many relation. """
+    _name = "mail.test.document.tag"
+    _description = "Tag"
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(default=True, string="Active")
+
+
+class MailTestDocumentFolder(models.Model):
+    """ Model required for mail.test.document to define a many2one relation. """
+    _name = "mail.test.document.folder"
+    _description = "Folder"
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(default=True, string="Active")
+
+
+class MailTestDocument(models.Model):
+    """ Model with all kind of relations and most field types to test with alias_defaults. """
+    _description = 'Document'
+    _name = 'mail.test.document'
+    _inherit = ['mail.thread']
```

Comment:
If we want to use it in some more test, we could add the blacklist / activity mixins :) as well as a company_id field, partner_id, email_from (that way we have something that looks like a classic mail thread enabled record)

Reply:
Indeed it is better if it can be reused, changed, thanks.